fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java

```diff
@@ -129,8 +129,12 @@ public void onSuccess(String eventName, Cursor cursor) throws IOException {
             } else if (status == 200) {
                 LOG.warn("Cursor for subscription [{}] to event [{}] in partition [{}] with offset [{}] was already committed", subscriptionId, eventName, cursor.getPartition(), cursor.getOffset());
             } else {
-                throw new IOException(String.format("Unexpected status code [%s] for subscription [%s] to event [%s]", status, subscriptionId, eventName));
+                throw new CursorOffsetCommitException(status, cursor, subscriptionId);
             }
+        } catch (IOProblem e) {
+            // in order to not have to refactor the entire ProblemHandlingRequest class,
+            // we are catching the error here and re-throwing a more specific one
+            throw new CursorOffsetCommitException(e.getStatusCode(), cursor, subscriptionId, e);
```

Comment:
worth mentioning exception in the method signature, because at the moment it worths IOException, but I do not see where exactly it is thrown